fakestorage/upload.go

```diff
@@ -391,9 +395,16 @@ func (s *Server) resumableUpload(bucketName string, r *http.Request) jsonRespons
 	}
 	s.uploads.Store(uploadID, obj)
 	header := make(http.Header)
-	header.Set("Location", s.URL()+"/upload/resumable/"+uploadID)
+	location := fmt.Sprintf(
+		"%s/upload/storage/v1/b/%s/o?uploadType=resumable&name=%s&upload_id=%s",
+		s.URL(),
+		url.PathEscape(bucketName),
```

Comment:
Nit: I assume that the bucket doesn't really need to be url-encoded? See https://cloud.google.com/storage/docs/naming-buckets
```suggestion
		bucketName,
```